### PR TITLE
Added new columns for Users table and Added new Access Request table

### PR DESCRIPTION
**Pull request by habibaz**
This pull request contains:

- Adding to new fields in user table [updated entity class and created migration file]

- Creating of new Access request table 

     1. create new entity
     2. create migration file
        - create new table, 
        - add trigger for updated_at fields
        - add trigger for creating a new user when request status changed to Grated.

### CODE REVIEW

File: src/migration/1694660281384-migration.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            `CREATE TYPE ${schemaName}.access_request_status AS ENUM ('NotGranted', 'Granted', 'Rejected');`,
+        );
+
+        // create request table
+        await queryRunner.query(`CREATE TABLE IF NOT EXISTS ${schemaName}."access_request" (
+        request_id UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
+        user_guid VARCHAR(36) NOT NULL,
+        identity_type VARCHAR(10) NOT NULL,
+        requested_role ${schemaName}.role_type NOT NULL,
+        organization VARCHAR(50) NOT NULL,
```
Comment (hannah-macdonald1) [inline src/migration/1694660281384-migration.ts:17]: They wouldn't neccessarily have an organization if they were BCGov, right?

File: src/migration/1694660281384-migration.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        // Add triggers for update at to table
+        await queryRunner.query(
+            `DROP TRIGGER IF EXISTS update_access_request ON ${schemaName}.access_request;`,
+        );
+        await queryRunner.query(
+            `CREATE TRIGGER update_access_request BEFORE UPDATE ON ${schemaName}.access_request FOR EACH ROW EXECUTE PROCEDURE ${schemaName}.update_column();`,
+        );
+
+        // Add trigger for access granted to create a new user
+        await queryRunner.query(`CREATE OR REPLACE FUNCTION ${schemaName}.add_new_user()
```
Comment (hannah-macdonald1) [inline src/migration/1694660281384-migration.ts:39]: Do we need to have this trigger also delete the request from the table if it was granted?

**Review by hannah-macdonald1 — COMMENTED** [review]
Looks good, I just have a few questions about the second migration file

File: src/migration/1694660281384-migration.ts (hunk 1 above)
Comment (habibaz) [reply src/migration/1694660281384-migration.ts:17]: Yup they will . We have an editable text field which is a required field for identity type IDIR and read only for BCeID.

**Review by hannah-macdonald1 — APPROVED** [review]
Looks good 👍

**Pull request merged**